### PR TITLE
Port current-column

### LINKED ISSUE #327: Port current-column to Rust
```c
// Defined in ~/src/emacs/src/indent.c
DEFUN ("current-column", Fcurrent_column, Scurrent_column, 0, 0, 0,
       doc: /* Return the horizontal position of point.  Beginning of line is column 0.
This is calculated by adding together the widths of all the displayed
representations of the character between the start of the previous line
and point (e.g., control characters will have a width of 2 or 4, tabs
will have a variable width).
Ignores finite width of frame, which means that this function may return
values greater than (frame-width).
Whether the line is visible (if `selective-display' is t) has no effect;
however, ^M is treated as end of line when `selective-display' is t.
Text that has an invisible property is considered as having width 0, unless
`buffer-invisibility-spec' specifies that it is replaced by an ellipsis.  */)
  (void)
{
  Lisp_Object temp;
  XSETFASTINT (temp, current_column ());
  return temp;
}
```

This will require adding a declaration in remacs-sys for the `current_column` C function.

### PR DESCRIPTION
**Pull request by shaleh**
Had to expose C current_column for this to work.

Closes #327

### CODE REVIEW

File: rust_src/src/indent.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+/// greater than (frame-width). Whether the line is visible (if
+/// `selective-display' is t) has no effect; however, ^M is treated as
+/// end of line when `selective-display' is t. Text that has an
+/// invisible property is considered as having width 0, unless
+/// `buffer-invisibility-spec' specifies that it is replaced by an
+/// ellipsis.
+#[lisp_fn]
+pub fn current_column() -> LispObject {
+    return LispObject::from_natnum(unsafe { remacs_sys::current_column() });
+}
```
Comment (brotzeit) [inline rust_src/src/indent.rs]: Can you remove the return for consistency reasons.

**Comment by shaleh** [comment]
Ooops. I need `return` to be ugly colored in my emacs 😀

On October 24, 2017 4:45:00 AM PDT, brotzeit <notifications@github.com> wrote:
>brotzeit commented on this pull request.
>
>
>
>> +/// Beginning of line is column 0.
>+/// This is calculated by adding together the widths of all the
>+/// displayed representations of the character between the start of
>+/// the previous line and point (e.g., control characters will have a
>+/// width of 2 or 4, tabs will have a variable width). Ignores finite
>+/// width of frame, which means that this function may return values
>+/// greater than (frame-width). Whether the line is visible (if
>+/// `selective-display' is t) has no effect; however, ^M is treated as
>+/// end of line when `selective-display' is t. Text that has an
>+/// invisible property is considered as having width 0, unless
>+/// `buffer-invisibility-spec' specifies that it is replaced by an
>+/// ellipsis.
>+#[lisp_fn]
>+pub fn current_column() -> LispObject {
>+    return LispObject::from_natnum(unsafe {
>remacs_sys::current_column() });
>+}
>
>Can you remove the return for consistency reasons.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/Wilfred/remacs/pull/392#pullrequestreview-71486925

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

File: rust_src/src/indent.rs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+use lisp::LispObject;
+use remacs_macros::lisp_fn;
```
Comment (brotzeit) [inline rust_src/src/indent.rs]: You forgot to add //! Indentation functions. in the first line.

Comment (shaleh) [reply rust_src/src/indent.rs]: Ah. Ok.

**Pull request merged**